shop/src/utils/generateVariants.js

```diff
@@ -1,25 +1,42 @@
-import flatMap from 'lodash/flatMap'
+const combinationsOfElementAndArray = (element, arr) => {
+  let accumulator = []
+  if (arr.length === 0) {
+    accumulator = [...accumulator, [element, arr[0]]]
+  }
+  for (let i = 0; i < arr.length; i++) {
+    if (Array.isArray(arr[i])) {
+      accumulator = [...accumulator, [element, ...arr[i]]]
```

Comment:
nit: Think you could just use `accumulator.push` here instead of creating a new array instance

Reply:
Agreed